code/modules/projectiles/guns/ballistic.dm

```diff
@@ -70,24 +70,23 @@
 			to_chat(user, "<span class='notice'>There's already a magazine in \the [src].</span>")
 	if(istype(A, /obj/item/suppressor))
 		var/obj/item/suppressor/S = A
-		if(can_suppress)
-			if(!suppressed)
-				if(!user.transferItemToLoc(A, src))
-					return
-				to_chat(user, "<span class='notice'>You screw [S] onto [src].</span>")
-				suppressed = A
-				S.oldsound = fire_sound
-				S.initial_w_class = w_class
-				fire_sound = 'sound/weapons/gunshot_silenced.ogg'
-				w_class = WEIGHT_CLASS_NORMAL //so pistols do not fit in pockets when suppressed
-				update_icon()
-				return
-			else
-				to_chat(user, "<span class='warning'>[src] already has a suppressor!</span>")
-				return
-		else
+		if(!can_suppress)
 			to_chat(user, "<span class='warning'>You can't seem to figure out how to fit [S] on [src]!</span>")
 			return
+		if(!user.is_holding(src))
```

Comment:
I feel this should be moved above if(!can_suppress), just to keep it consistent that you have to be holding the weapon for suppressor interactions. Opinions?

Reply:
well then you're still doing the more expensive check first

Reply:
Fair enough.